### ISSUE TITLE
Database target: ProviderName is required when using <connectionstrings>

### ISSUE THREAD

**Issue by 304NotModified** [NLog/NLog#1115]
Providername attributes is required if using the database target in combination with `<connectionstings>`.  Skipping it, or using another value in the NLog configuration leads to an error. 

> If you are going to use a named connection in a configuration file, that connection must specify the System.Data.SqlClient namespace as its provider.

>That’s because NLog relies on the Entity Framework to configure its data objects; if your named connection does not represent itself as being provided by Sql.Data.SqlClient, then NLog cannot resolve the proper form of its data objects, and will throw an error such as:

>
Error initializing target Database Target[X] System.ArgumentException: Expecting non-empty string for 'providerInvariantName' parameter.
So your configuration file connection string entry, for whichever database is going to host your NLog records, should look something like this:

```xml
<configuration>
   <connectionStrings>
      <add name="DefaultConnection" providerName="System.Data.SqlClient" connectionString="Server=localhost;Database=mydatabase;User ID=username;Password=password;Trusted_Connection=False;Encrypt=True;Connection Timeout=30;" />
   </connectionStrings>
</configuration>
```

https://www.dougv.com/2015/12/using-sql-server-stored-procedures-nlog/

We should check if we can not throw an error and take the type from the NLog configuration

**Comment by 304NotModified** [comment]
https://github.com/NLog/NLog/blob/master/src/NLog/Targets/DatabaseTarget.cs#L332

**Comment by 304NotModified** [comment]
Fixed in NLog 4.3!
